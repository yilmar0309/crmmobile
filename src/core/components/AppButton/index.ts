export {AppButton, AppButtonVariant, AppButtonSizeVariant} from './AppButton';
