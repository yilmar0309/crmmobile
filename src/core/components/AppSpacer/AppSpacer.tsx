import React from 'react';
import {DimensionValue, View} from 'react-native';

interface IAppSpacer {
  height?: DimensionValue | undefined;
}

export const AppSpacer = ({height = 16}: IAppSpacer) => (
  <View testID="app-spacer" style={{height}} />
);
