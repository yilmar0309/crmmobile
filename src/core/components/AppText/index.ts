export {AppText, AppTextVariant} from './AppText';
