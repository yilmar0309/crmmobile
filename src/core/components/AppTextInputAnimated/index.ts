export {AppTextInputAnimated, TextInputState} from './AppTextInputAnimated';
