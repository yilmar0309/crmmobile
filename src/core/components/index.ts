export {AppText, AppTextVariant} from './AppText';
export {AppButton, AppButtonVariant, AppButtonSizeVariant} from './AppButton';
export {AppTextInputAnimated, TextInputState} from './AppTextInputAnimated';
export {AppContainer} from './AppContainer';
export {AppSpacer} from './AppSpacer';
export {AppHeaderLeftBack} from './AppHeaderLeftBack';
export {AppShowEmptyList} from './AppShowEmptyList';
export {AppLoading} from './AppLoading';
export {AppImage} from './AppImage';
export {AppLoadingList} from './AppLoadingList';
