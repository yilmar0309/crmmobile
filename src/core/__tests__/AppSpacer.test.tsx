import React from 'react';
import {render} from '@testing-library/react-native';
import {AppSpacer} from '../components/AppSpacer';

describe('AppSpacer', () => {
  it('renders correctly with default height', () => {
    const {getByTestId, toJSON} = render(<AppSpacer />);
    const spacer = getByTestId('app-spacer');
    expect(spacer).toBeTruthy();
    expect(spacer.props.style).toMatchObject({height: 16});
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies custom height when provided', () => {
    const customHeight = 50;
    const {getByTestId} = render(<AppSpacer height={customHeight} />);
    const spacer = getByTestId('app-spacer');
    expect(spacer.props.style).toMatchObject({height: customHeight});
  });

  it('handles undefined height correctly', () => {
    const {getByTestId} = render(<AppSpacer height={undefined} />);
    const spacer = getByTestId('app-spacer');
    // Expect default height to be used when height is undefined
    expect(spacer.props.style).toMatchObject({height: 16});
  });
});
