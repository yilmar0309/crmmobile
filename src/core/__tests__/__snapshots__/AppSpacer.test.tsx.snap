// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSpacer renders correctly with default height 1`] = `
<View
  style={
    {
      "height": 16,
    }
  }
  testID="app-spacer"
/>
`;
